packages/xo-web/src/common/form/select.js

```diff
@@ -142,15 +142,15 @@ export default class Select extends React.PureComponent {
       simpleValue,
       value,
     } = props
+    let option
     if (
       autoSelectSingleOption &&
       options != null &&
-      options.length === 1 &&
+      ([option] = options.filter(_ => !_.disabled)).length === 1 &&
```

Comment:
`filter`'s complexity is `O(n)` do you think it's a good idea to do it at every update?

Maybe you could reorder the test to do it less often?